spec/models/user_spec.rb

```diff
@@ -13,39 +13,14 @@
     expect(user).to be_valid
   end
 
-  it "is invalid without a username" do
-    user = User.new(username: nil)
-    user.valid?
-    expect(user.errors[:username]).to include("can't be blank")
-  end
-
-  it "is invalid without an email address" do
-    user = User.new(email: nil)
-    user.valid?
-    expect(user.errors[:email]).to include("can't be blank")
-  end
-
-  it "is invalid without a password" do
-    user = User.new(password_digest: nil)
-    user.valid?
-    expect(user.errors[:password_digest]).to include("can't be blank")
-  end
-
-  it "is invalid with a duplicate email address" do
-    User.create(
-      username: "Bob77",
-      password_digest: "abcd1234",
-      email: "bob77@testemail.com"
-    )
-
-    user = User.new(
-      password_digest: "onetwothree",
-      email: "bob77@testemail.com"
-    )
-
-    user.valid?
-    expect(user.errors[:email]).to include("has already been taken")
-  end
+  it { should validate_presence_of(:username) }
+  it { should validate_presence_of(:email) }
+  it { should validate_presence_of(:password_digest) }
+
+  subject { User.new(email: 'foo@bar.com', password_digest: 'password', username: 'foo') }
```

Comment:
I'd probably just have this at the top. It's weird to separate the validations this way, even if there's a technical requirement.